Added ability to create tasks in folders

dibenzepin opened the pull request:
## Added:
- `ScheduleBuilder<Frequency>.in_folder()` method
- `ScheduleGetter` struct

## Updated:
- Examples
- `windows` crate

## Remarks

I wanted to try to use this crate in a project but needed to be able to put the tasks in a specified folder so that they would be easier to manage later on, so I tried adding that functionality with an `in_folder()` method on `ScheduleBuilder`. It works by checking if the folder already exists, and creating it if it doesn't, then putting the task into that folder.

I also took the liberty of updating the `windows` crate to the newest version, and fixing all the errors that came because of that (mostly just some structs moving around, adding a reference that wasn't needed before, `.into()` not inferring the correct type)

`ScheduleGetter` is for the purpose of being able to read tasks after they've been added, but I'm not quite sure if this functionality should be in this crate or another. Right now it just returns an `IRegisteredTask` as opposed to an actual `Schedule` so it still depends on `windows` to be of any use.

dibenzepin committed:
fixed examples

dibenzepin committed:
updated windows crate to `0.48.0`

dibenzepin committed:
added `in_folder()` to schedule builder

dibenzepin committed:
Add schedule_getter struct

mattrobineau commented:
> I wanted to try to use this crate in a project but needed to be able to put the tasks in a specified folder so that they would be easier to manage later on, so I tried adding that functionality with an `in_folder()` method on `ScheduleBuilder`. It works by checking if the folder already exists, and creating it if it doesn't, then putting the task into that folder.

Thank you for taking an interest in improving `planif`. I think that the users of `planif` would find being able to specify a folder for tasks useful. 
 
> `ScheduleGetter` is for the purpose of being able to read tasks after they've been added, but I'm not quite sure if this functionality should be in this crate or another. Right now it just returns an `IRegisteredTask` as opposed to an actual `Schedule` so it still depends on `windows` to be of any use.

"In the beginning".. I had thought about whether or not to add the ability to read tasks and then modify them. At the time I had decided reading and/or modifying existing tasks from the Windows Task Scheduler to be outside of the scope for the MVP (Minimum Viable Product).

At this time I believe a case could be made to add this functionality to the crate. I would, however, request that this feature be in its own PR. To be transparent, I don't think the current iteration of `ScheduleGetter` fits well with the crate's design but if changes were brought to the implementation I wouldn't object to adding the functionality to the crate.

If you could please remove the `ScheduleGetter` implementation in this PR. Afterwards, if you are up for it, you can open a new PR for `SchedulerGetter` and we could discuss the specifics of making it fit with the overall design of `planif`.

Again, thank you @poopsicles for your work and time.

dibenzepin committed:
Revert "Add schedule_getter struct"

This reverts commit c484c718ab696c453aba112b6630245f412bde71.

dibenzepin commented:
Thanks for replying @mattrobineau, I really appreciate it.
Alright, I've reverted my `ScheduleGetter` commit and moved it to a new branch...I'll make another PR shortly
